Correctly parse suggestions containing multiple words

Hopper262 opened the pull request:
When not using the "--run-together" argument, aspell will treat compound words like "webdesign" as a misspelling, and provide "web design" as the first suggestion. This is not parsed correctly by the current code, and the suggestions list starts with "web" and "design" as two separate items. This PR re-parses the suggestions part of the line, splitting only on commas and not simply whitespace, to deliver multiple-word suggestions as a single item as intended.

Thanks for this module, btw -- still works great under Node 16!

Hopper262 committed:
Correctly parse suggestions containing multiple words

xavi- commented:
Hello!  Thanks the PR.

Unfortunately, I haven't been able to reproduce the error on my end.  Is there any chance you could send over the raw aspell output?  I.e, the output from this command:

```
$ echo "problematic text" | aspell -a
```

If it's helpful, I've just added some unit tests to the project

Hopper262 committed:
Merge branch 'xavi-:master' into run-together-suggestions

Hopper262 committed:
Parse first alternative correctly; ignore alternatives for '#' lines

Hopper262 committed:
Update test for alternatives with spaces

Hopper262 commented:
Hi, thanks much for the unit tests! The typo test actually includes an example of this problem:

* In the aspell output, there are 11 alternatives separated by commas. The third alternative is "miss take" with two words separated by a space.
* In the node-aspell `alternatives` array, there are 12 alternatives instead of 11. The third is "miss" and the fourth is "take" which I would only expect if the aspell output was "..., miss, take, ..." instead.

Another example is the command `echo "webdesign" | aspell -a` where "web design" should be treated as the first alternative. My use case is a search query "did you mean ..." feature, and it's not correct to replace "webdesign" with just "web" without the second word "design" associated with it.

I updated the test in this PR to reflect the expected output with this fix, and also fixed some dumb bugs in my first attempt. There's probably a nicer way to parse the output, but hopefully this makes the issue clearer.

xavi- commented:
Ahh, I understand now.  Good catch.  Thanks for the explanation!

xavi- commented:
Fix published in [`0.1.1`](https://www.npmjs.com/package/aspell/v/0.1.1)
